### PR TITLE
basic/mountpoint-util: whitelist 'exfat' in fstype_can_uid_gid

### PR DESCRIPTION
**Pull request by floppym**

### CODE REVIEW

File: src/basic/mountpoint-util.c [hunk 1]
```diff
@@ -359,6 +359,7 @@ bool fstype_can_uid_gid(const char *fstype) {
 
         return STR_IN_SET(fstype,
                           "adfs",
+                          "exfat",
```
Comment (poettering) [inline src/basic/mountpoint-util.c:362]: is that an upstream fs? i.e. in the regular kernel?

Comment (floppym) [reply src/basic/mountpoint-util.c:362]: There are 2 implementations:

- exfat (FUSE) https://github.com/relan/exfat
- exfat-nofuse (kernel mod) https://github.com/dorimanx/exfat-nofuse

Both the FUSE-based and kernel module support the uid and gid options. The exFAT filesystem spec does not include file ownership or permissions.

Comment (floppym) [reply src/basic/mountpoint-util.c:362]: To more directly answer the question: no, there is no implementation in the mainline kernel due to license and/or patent issues.

Comment (poettering) [reply src/basic/mountpoint-util.c:362]: Is there any perspective for this to go upstream? I through microsoft donated the relevant patents to OIN?

I am a bit concerned about adding these entries for stuff that has no perspective to ever going upstream and being officially supported upstream...

Comment (floppym) [reply src/basic/mountpoint-util.c:362]: > Is there any perspective for this to go upstream? I through microsoft donated the relevant patents to OIN?

I am not sure.

> I am a bit concerned about adding these entries for stuff that has no perspective to ever going upstream and being officially supported upstream...

I don't really understand this concern. I don't think exfat is going to disappear anytime soon.

The existing FUSE implementation is supported on several major distros, including Debian. [removed Fedora; I mistook a Google result for a third-party package]

In the worst, very unlikely case, a third exfat implementation might be merged into the mainline kernel without supporting the uid/gid options. I just can't see that happening.

Comment (keszybz) [reply src/basic/mountpoint-util.c:362]: Yeah, I think it's OK to put an established-even-if-not-upstream filesystem here. The maintenance "cost" is negligible on our side, but it makes integration with such file systems much nicer for the users.

**Comment by poettering** [comment]
Well. This really smells like something that will bitrot badly. If this ever goes upstream I figure it would be folded into the existing vfat driver (and thus needs no support from our side). I am not sure its really a good idea merging a stopgap like this which has no clear future and if it has one looks different and incompatible with what this patch does.

**Comment by poettering** [comment]
At the very least this deserves a comment in the code explaining the situation. But quite frankly i would not merge this at all. In particular as there are multiple implementations around for this and the fuse one isnt even covered by this patch.

**Comment by floppym** [comment]
> In particular as there are multiple implementations around for this and the fuse one isnt even covered by this patch.

I'm not sure what you mean by that. libblkid returns "exfat", and it is common practice to symlink `mount.exfat` to `mount.exfat-fuse` so that auto-mounting will work.

**Comment by floppym** [comment]
In case it wasn't clear, my goal here is to make `systemd-mount --user floppym /dev/sdx` work properly with an exfat volume. Previously it would output an incorrect error message stating that exfat does not support the uid option.

If you're going to revert this, please come up with a better way of detecting uid/gid than a hard-coded list of filesystem names.

**Pull request merged**